docs/001_develop/03_client-capabilities/008_interaction/checkbox.mdx

```diff
@@ -0,0 +1,180 @@
+tags:
+- interaction
+- interactive
+- checkbox
+- box
+- boolean
+- form
+- click
+- change
+- checked
+- toggle
+- submit
+- data
+sidebar_position: 8
+---
+
+import CheckboxDemo from '../../../../examples/ui/client-capabilities/interaction/checkbox.js';
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Checkbox
+
+A form-connected implementation of a [checkbox](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox) input.
+
+Use cases:
+* Multi-selection forms
+* Filtering
+* Sorting
+* To-Do lists
+* Privacy and consent
+
+Any use where the user is to give a boolean type input choice.
+
+## Example
+
+<CheckboxDemo />
+
+<Tabs defaultValue="genesis" values={[{ label: 'Genesis', value: 'genesis', }, { label: 'React', value: 'react', }, { label: 'Angular', value: 'angular', }]}>
+
+<TabItem value="genesis">
+
+Declaration
+```typescript
+<rapid-checkbox></rapid-checkbox>
+```
+Usage
+```typescript
+@customElement({
+  name: 'my-element',
+  template: html`
+        <rapid-checkbox
+            ?disabled="${(x) => x.disabled}"
+            ?checked="${sync((x) => x.isSelected, 'boolean')}"
+            @change="${(x) => x.checkChanged()}"
+        >Checkbox
+        </rapid-checkbox>
+    `,
+})
+export class MyElement extends GenesisElement {
+  @observable disabled = false;
+  @observable isSelected = false;
+  checkChanged() {
+    console.log(this.isSelected);
+  }
+}
+```
+
+</TabItem>
+<TabItem value="react">
+
+Declaration
+```typescript
+<rapid-checkbox></rapid-checkbox>
+```
+Usage
+```typescript
+export function MyComponent() {
+	const [checkboxValue, setCheckboxState] = useState(false);
+	const handleCheckboxChange = (event) => {
+		setCheckboxState(!!event.target.checked);
+		console.log({ checkboxValue })
+	};
+
+	return (
+		<rapid-checkbox onChange={handleCheckboxChange}></rapid-checkbox>
+	);
+}
+```
+</TabItem>
+<TabItem value="angular">
+
+Declaration
+```typescript
+<rapid-checkbox></rapid-checkbox>
+```
+
+Model
+```typescript my.component.ts
+import { Component, CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
+import { FormsModule } from '@angular/forms';
+@Component({
+  selector: 'my-root',
+  templateUrl: './my.component.html',
+  styleUrls: ['./my.component.css'],
+  standalone: true,
+  schemas: [CUSTOM_ELEMENTS_SCHEMA],
+  imports: [FormsModule],
+})
+
+export class AppComponent {
+  isSelected: boolean = false;
+  onCheckboxChange(event: Event) {
+    const target = event.target as HTMLInputElement;
+    this.isSelected = target.checked;
+    console.log(this.isSelected);
+  }
+}
+```
+View
+```html
+<rapid-checkbox
+name="isSelected"
+[checked]="isSelected"
+(change)="onCheckboxChange($event)"
+></rapid-checkbox>
+```
+</TabItem>
+</Tabs>
+
+## DOM API
+
+Property and attribute binding examples for Genesis Component syntax. Closing tag omitted.
+
+### Attributes
+
+| Attribute | Type | Use | Example |
+|---|---|---|---|
```

Comment:
In your specific case the property would be `someAttribute` and the attribute would be `some-attribute`. I have listed both for completeness - I assume we want the user to use attrivutes where possible, but listing the properties is useful for someone accessing via javascript rather than the DOM.

Either way I don't think that applies to this PR specifically

Reply:
I don't think we should document properties that have attributes associated to them. If you want to do that using javascript you can use https://developer.mozilla.org/en-US/docs/Web/API/Element/setAttribute